src/antlr4/Cypher.g4

```diff
@@ -334,8 +334,8 @@ oC_StringListNullOperatorExpression
     : oC_PropertyOrLabelsExpression ( oC_StringOperatorExpression | oC_ListOperatorExpression | oC_NullOperatorExpression )? ;
 
 oC_ListOperatorExpression
-    : ( SP ? '[' oC_Expression ']' )
-        | ( SP ? '[' oC_Expression? ':' oC_Expression? ']' );
+    : ( ( SP ? '[' oC_Expression ']' )
```

Comment:
this rule is getting complicates so I would try to rewrite as
```
oC_ListOperatorExpression
    : ( kU_ListExtractOperatorExpression | kU_ListSliceOperatorExpression ) oC_ListOperatorExpression ? ;

kU_ListExtractOperatorExpression
   :  SP ? '[' oC_Expression ']' ;

kU_ListSliceOperatorExpression
   :  SP ? '[' oC_Expression? ':' oC_Expression? ']' ;

```

Let me know if this makes sense or not

Reply:
Done in [organized parser rules](https://github.com/kuzudb/kuzu/pull/1054/commits/9f3ba9ed3e3f304b4357a7353a8d26287a094dba)